core/src/main/java/org/infinispan/interceptors/impl/PrefetchInterceptor.java

```diff
@@ -207,68 +214,97 @@ private Object prefetchKeysIfNeededAndInvokeNext(InvocationContext ctx, Visitabl
                nil -> makeStage(prefetchKeysIfNeededAndInvokeNext(ctx, command, keys, true)).toCompletableFuture()));
       }
       if (transferedKeys != null) {
-         return asyncInvokeNext(ctx, command, retrieveRemoteValues(ctx, transferedKeys).toCompletableFuture());
+         return asyncInvokeNext(ctx, command, retrieveRemoteValues(ctx, transferedKeys, command.getTopologyId()).toCompletableFuture());
       } else {
          return invokeNext(ctx, command);
       }
    }
 
-   private InvocationStage retrieveSingleRemoteValue(InvocationContext ctx, Object key) {
+   private InvocationStage retrieveSingleRemoteValue(InvocationContext ctx, Object key, int topologyId) {
```

Comment:
That (naming) wouldn't be right: it invokes the command *and* then does the remote lookup. The returned InvocationStage is completed when both those complete.